src/ripple/app/paths/impl/StepChecks.h

```diff
@@ -0,0 +1,95 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_APP_PATHS_IMPL_STEP_CHECKS_H_INCLUDED
+#define RIPPLE_APP_PATHS_IMPL_STEP_CHECKS_H_INCLUDED
+
+#include <ripple/basics/Log.h>
+#include <ripple/ledger/ReadView.h>
+#include <ripple/protocol/AccountID.h>
+#include <ripple/protocol/UintTypes.h>
+
+#include <beast/beast/utility/Journal.h>
+
+namespace ripple {
+
+inline
+TER
+checkFreeze (
+    ReadView const& view,
+    AccountID const& src,
+    AccountID const& dst,
+    Currency const& currency)
+{
+    if (src != dst)
+    {
+        // check freeze
+        if (auto sle = view.read (keylet::account (dst)))
+        {
+            if (sle->isFlag (lsfGlobalFreeze))
+            {
+                return terNO_LINE;
+            }
+        }
+
```

Comment:
`src == dst` should not happen. Book steps do not use this check (book steps see frozen offer owners as having a zero balance). It doesn't make sense to have a direct step with the same src and dst. I changed the `if` statement to an assert.